### DISCUSSION TITLE
Is there a way to use contentFunction in purgecss

### DISCUSSION THREAD

**Discussion by MateBenko** [tailwindlabs/tailwindcss#4085] [Help]
Hey there,

We've seen in the documentation of postcss-purgecss that is supports contentFunction. 

Our question is how can we use this property throught tailwind config? We would need this so we could purge the css individually for every web component.

Thanks for your reply in advance.

**Comment by dance2die** [comment]
You can pass contentFunction in the purge css option.
https://tailwindcss.com/docs/optimizing-for-production#purge-css-options

The doc says you can pass PostCSS options listed here.
https://purgecss.com/plugins/postcss.html#usage

```js
module.exports = {
  purge: {
    content: ['./src/**/*.html'],

    // These options are passed through directly to PurgeCSS
    options: {
      contentFunction: (sourceFile) => {
        console.log({ sourceFile });
      }
    },
  },
  // ...
}
```
I am sorry, I don't know how to use `contentFunction` so this is all I can show... :sweat:

**Reply by simonswiss** [reply]
Correct! Any option you put in `purge.options` will be passed straight to Purge CSS, so you can interface with their entire API through this "escape hatch" 👍
